### PR TITLE
Generate api_key and assign to user.

### LINKED ISSUE #28: generate secure api_key

### PR DESCRIPTION
**Pull request by bayendor**
Includes a method `check_api_key` in users api controller.

Closes #28

### CODE REVIEW

File: app/models/api_key.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+class ApiKey < ActiveRecord::Base
+  belongs_to :user
+  before_create :generate_access_token
+
+  validates :access_token, uniqueness: true
+
+  private
+
+  def generate_access_token
```
Comment (dalexj) [inline app/models/api_key.rb]: do this instead?

```ruby
self.access_token = nil
  while access_token && self.class.exists?(access_token: access_token)
    self.access_token = SecureRandom.base64(12)
  end
```

File: app/controllers/api/users_controller.rb [hunk 2]
```diff
@@ -20,4 +21,9 @@ def merge_in_new_commands(user)
   def big_bro_params
     JSON.parse(params["big_bro"])
   end
+
+  def check_api_key
+    api_key = ApiKey.where(access_token: params[:access_token])
```
Comment (dalexj) [inline app/controllers/api/users_controller.rb]: `.where` or `.find_by`?

where will return the keys that have that acess_token, find will just take out the single one

File: app/controllers/api/users_controller.rb [hunk 3]
```diff
@@ -20,4 +21,9 @@ def merge_in_new_commands(user)
   def big_bro_params
     JSON.parse(params["big_bro"])
   end
+
+  def check_api_key
```
Comment (bayendor) [inline app/controllers/api/users_controller.rb]: Could be used like this in the HTTP header: 

`$ curl http://localhost:3000/api/products -H 'Authorization: Token token="afbadb4ff8485c0adcba486b4ca90cc4"'`

File: app/models/api_key.rb (hunk 1 above)
Comment (bayendor) [inline app/models/api_key.rb]: Agreed, refactored the method.

**Pull request merged**